### PR TITLE
[Enhancement] use map instead of List<pair> as cloud properties

### PR DESCRIPTION
**Pull request by stephen-shelby**
Fixes #issue
Community users want to pass some custom parameters. They said iterating over a vector is cumbersome. This pr is to use map instead of vector processing cloud properties.

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: gensrc/thrift/CloudConfiguration.thrift [hunk 1]
```diff
@@ -23,12 +23,7 @@ enum TCloudType {
...
 }
 
-struct TCloudProperty {
-    1: required string key;
-    2: required string value;
-}
-
 struct TCloudConfiguration {
     1: optional TCloudType cloud_type;
-    2: optional list<TCloudProperty> cloud_properties;
```
Comment (imay) [inline gensrc/thrift/CloudConfiguration.thrift:33]: This change will cause compatibility issues.
The process who is running new binary can not recognize the older version's requests.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 24 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/credential/aliyun/AliyunCloudCredential.java|0|3|00.00%|[56, 57, 58]|
|:large_blue_circle:|com/starrocks/credential/aws/AWSCloudConfiguration.java|0|4|00.00%|[72, 73, 74, 75]|
|:large_blue_circle:|com/starrocks/credential/azure/AzureCloudConfiguration.java|0|1|00.00%|[39]|
|:large_blue_circle:|com/starrocks/credential/azure/AzureStorageCloudCredential.java|0|1|00.00%|[53]|
|:large_blue_circle:|com/starrocks/credential/gcp/GCPCloudCredential.java|0|1|00.00%|[86]|
|:large_blue_circle:|com/starrocks/credential/aliyun/AliyunCloudConfiguration.java|0|2|00.00%|[43, 44]|
|:large_blue_circle:|com/starrocks/credential/gcp/GCPCloudConfiguration.java|0|1|00.00%|[41]|
|:large_blue_circle:|com/starrocks/credential/aws/AWSCloudCredential.java|0|11|00.00%|[228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238]|

File: gensrc/thrift/CloudConfiguration.thrift (hunk 1 above)
Comment (stephen-shelby) [reply gensrc/thrift/CloudConfiguration.thrift:33]: Done, I have tested it using old fe and new be.

File: gensrc/thrift/CloudConfiguration.thrift [hunk 2]
```diff
@@ -31,4 +31,5 @@ struct TCloudProperty {
 struct TCloudConfiguration {
     1: optional TCloudType cloud_type;
     2: optional list<TCloudProperty> cloud_properties;
```
Comment (Smith-Cruise) [inline gensrc/thrift/CloudConfiguration.thrift:33]: add some comments, let people know it should be deprecated.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 28 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/credential/aliyun/AliyunCloudCredential.java|0|3|00.00%|[56, 57, 58]|
|:large_blue_circle:|com/starrocks/credential/aws/AWSCloudConfiguration.java|0|5|00.00%|[72, 73, 74, 75, 77]|
|:large_blue_circle:|com/starrocks/credential/azure/AzureCloudConfiguration.java|0|2|00.00%|[39, 41]|
|:large_blue_circle:|com/starrocks/credential/azure/AzureStorageCloudCredential.java|0|1|00.00%|[53]|
|:large_blue_circle:|com/starrocks/credential/gcp/GCPCloudCredential.java|0|1|00.00%|[86]|
|:large_blue_circle:|com/starrocks/credential/aliyun/AliyunCloudConfiguration.java|0|3|00.00%|[43, 44, 46]|
|:large_blue_circle:|com/starrocks/credential/gcp/GCPCloudConfiguration.java|0|2|00.00%|[41, 43]|
|:large_blue_circle:|com/starrocks/credential/aws/AWSCloudCredential.java|0|11|00.00%|[228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 28 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/credential/aliyun/AliyunCloudCredential.java|0|3|00.00%|[56, 57, 58]|
|:large_blue_circle:|com/starrocks/credential/aws/AWSCloudConfiguration.java|0|5|00.00%|[72, 73, 74, 75, 77]|
|:large_blue_circle:|com/starrocks/credential/azure/AzureCloudConfiguration.java|0|2|00.00%|[39, 41]|
|:large_blue_circle:|com/starrocks/credential/azure/AzureStorageCloudCredential.java|0|1|00.00%|[53]|
|:large_blue_circle:|com/starrocks/credential/gcp/GCPCloudCredential.java|0|1|00.00%|[86]|
|:large_blue_circle:|com/starrocks/credential/aliyun/AliyunCloudConfiguration.java|0|3|00.00%|[43, 44, 46]|
|:large_blue_circle:|com/starrocks/credential/gcp/GCPCloudConfiguration.java|0|2|00.00%|[41, 43]|
|:large_blue_circle:|com/starrocks/credential/aws/AWSCloudCredential.java|0|11|00.00%|[228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238]|

**Review by Youngwb — APPROVED** [review]

**Review by Smith-Cruise — APPROVED** [review]

**Review by miomiocat — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by Smith-Cruise** [comment]
@Mergify backport branch-3.0

**Pull request merged**